packages/interactivity-router/src/index.js

```diff
@@ -128,23 +189,34 @@ export const { state, actions } = store( 'core/router', {
 		 * needed, and updates any interactive regions whose contents have
 		 * changed. It also creates a new entry in the browser session history.
 		 *
-		 * @param {string}  href                               The page href.
-		 * @param {Object}  [options]                          Options object.
-		 * @param {boolean} [options.force]                    If true, it forces re-fetching the URL.
-		 * @param {string}  [options.html]                     HTML string to be used instead of fetching the requested URL.
-		 * @param {boolean} [options.replace]                  If true, it replaces the current entry in the browser session history.
-		 * @param {number}  [options.timeout]                  Time until the navigation is aborted, in milliseconds. Default is 10000.
-		 * @param {boolean} [options.loadingAnimation]         Whether an animation should be shown while navigating. Default to `true`.
-		 * @param {boolean} [options.screenReaderAnnouncement] Whether a message for screen readers should be announced while navigating. Default to `true`.
+		 * @param {string|Event} eventOrUrl                         The page href or the event handler in case it is used directly in a directive.
+		 * @param {Object}       [options]                          Options object.
+		 * @param {boolean}      [options.force]                    If true, it forces re-fetching the URL.
+		 * @param {string}       [options.html]                     HTML string to be used instead of fetching the requested URL.
+		 * @param {boolean}      [options.replace]                  If true, it replaces the current entry in the browser session history.
+		 * @param {number}       [options.timeout]                  Time until the navigation is aborted, in milliseconds. Default is 10000.
+		 * @param {boolean}      [options.loadingAnimation]         Whether an animation should be shown while navigating. Default to `true`.
+		 * @param {boolean}      [options.screenReaderAnnouncement] Whether a message for screen readers should be announced while navigating. Default to `true`.
 		 *
 		 * @return {Promise} Promise that resolves once the navigation is completed or aborted.
 		 */
-		*navigate( href, options = {} ) {
+		*navigate( eventOrUrl, options = {} ) {
```

Comment:
Having `eventOrUrl` here is not great from an API perspective. Where's `navigate()` used with a URL? I would suggest we implement a separate action for that, e.g. `navigateUrl()` or `navigateToUrl()`.

This doesn't mean we have to duplicate all the code. We can probably have a helper function with the reusable bits somewhere, or maybe one action could even call the other.

Reply:
In [this commit](https://github.com/WordPress/gutenberg/pull/59707/commits/d72cd023eae40645e8e787f24d91f515b12ce4ee), I changed the strategy and it registers the event handlers in JS instead of using directives. That allow us to reuse the existing navigate and prefetch functions as they are today.